x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -332,44 +336,76 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
             },
           },
         },
+        size: 0,
         aggs: {
           id: {
             terms: {
               field: 'monitor.id',
+              size: 1000,
+            },
+            aggs: {
+              latest: {
+                top_hits: {
+                  size: 1,
+                  sort: [
+                    {
+                      '@timestamp': {
+                        order: 'desc',
+                      },
+                    },
+                  ],
+                },
+              },
             },
           },
           port: {
             terms: {
               field: 'url.port',
+              size: 1000,
             },
           },
-          scheme: {
+          status: {
```

Comment:
Since we're looking for _values_ rather than counts, we don't even need to look this up.